### DISCUSSION TITLE
Report for LF Mentorship - CodeChecker

### DISCUSSION THREAD

**Discussion by jay24rajput** [Ericsson/codechecker#3239] [Show and tell]
Hello, 

My name is Jay Rajput and I have been working on Codechecker for the past 6 months as Linux Foundation Mentee with @bulwahn as my mentor. I have worked on extending report converter for tools such as coccinelle, match, sparse, kernel-doc, sphinx. I have also developed commands for exporting and importing the comments and review status via CodeChecker CLI. 

 We have now completed the goals decided upon successfully. We have also developed a detailed report on the work done in the mentorship. We would like you to have a look at the same and if you have any CodeChecker blogs or channels, publish the report on those mediums 

Here is the link to the report: https://docs.google.com/document/d/1g4OhTaDtsCG0zR8M3QsRFVY54osshj57mT0N0RM0HCw/edit?usp=sharing

**Comment by Xazax-hun** [comment]
Nice work! Thanks for all your valuable contributions, and hope to see you around :)

**Reply by jay24rajput** [reply]
Thanks a lot 😄

**Comment by csordasmarton** [comment]
@jay24rajput Thank you very much for your hard work what you did in CodeChecker :blush:. You did a really great job. You helped us to improve CodeChecker with many new features, you developed your patches and fixed review comments really fast and you showed us what the open source community means!

Also thank you very much for @bulwahn who mentored you and helped us in the review process and brainstorming.

Thank you very much guys that you shared your experiences with us and we hope that you liked it and we wish you all the best. We hope, that if you have free time you may return contributing :blush:.

**Comment by dkrupp** [comment]
Thanks @jay24rajput  for your hard work and excellent contributions!

Just out of curiousity: In which kernel development project is CodeChecker being used? 
Do you have an open link to the database maybe?

Thanks again and hope to see you around! :)

**Reply by jay24rajput** [reply]
I was just actually an intern. But I know it is definitely being used by [Elisa project](https://elisa.tech/). Maybe @bulwahn can answer the question more appropriately.

**Reply by bulwahn** [reply]
We are experimenting with CodeChecker in a small group called Tool Investigation and Code Improvement subgroup in the Elisa Project, https://elisa.tech. We would like to understand and work on the topic how tracking static analysis findings could be effectively made an integral part of the kernel development and its practices: kernel development includes thousands of contributors, hundreds of changes each day and a very special development process based on email and a number of scripts. So, there is still a lot of work required to make CodeChecker fit for this community at large. But of course, CodeChecker is already usable and valuable for a smaller group to work with.

You can find our CodeChecker instance at https://elisa-builder-00.iol.unh.edu:8001/ (sorry, we do not have a properly signed SSL certificate yet; it is on our TODO list.). If you would like to have access, please just contact me or even better just contact @sudipm-mukherjee. We currently track around 12,000 findings with a new run each day and we would like to assess the new findings in the repository every day. At the moment, the CodeChecker diff reports roughly 100 findings being resolved and 100 new findings between two runs among those 12,000 findings, but my initial assessment is that the heuristics for tracking the already known findings just needs some further fine-tuning and then we should be able to determine the actual new findings much more precisely and hence make that task of tracking new findings much more manageable by a small group. We can probably discuss many of the details of features required in separate issues in this issue tracker here.

Anyway, our experiment and journey with CodeChecker will continue! Thanks for all your assistance and thanks to everyone who has contributed to CodeChecker sofar.
